### ISSUE TITLE
Pandas packing on Windows missing necessary .dlls

### ISSUE THREAD

**Issue by bjones1** [pyinstaller/pyinstaller#1969]
This is a followup on #1580. On my Windows PC (see below), I see failures in `tests/functional/test_libraries.py::test_pandas_extension[onedir]` (and possibly onefile -- I'm solving one failure at a time).  Investigating, running test test reports `Intel MKL FATAL ERROR: Cannot load mkl_intel_thread.dll`. Copying from `site-packages\numpy\core\*.dll` to the `dist\` directory fixes this. (Just copying the requested `.dll` by itself doesn't.)

I'm assuming that PyInstaller can't analyze DLL dependencies -- @matysek, can you confirm or deny? Based on this, I can write a Windows-specific hook. Does anyone see this failure on other platforms (meaning hooks for other OSes)?

Version info:
```
__main__.py                 80 INFO     PyInstaller: 3.2.dev0+g5e80840.mod
__main__.py                 81 INFO     Python: 3.4.4
__main__.py                 82 INFO     Platform: Windows-10-10.0.10240
```

**Comment by bjones1** [comment]
Just looking at  the log answers my question: yes, PyInstaller does walk DLL dependencies. For example:

````
bindepend.py               211 DEBUG    Analyzing c:\users\bjones\downloads\winpython-32bit-3.4.4.1qt5\python-3.4.4\lib\site-packages\numpy\core\mkl_rt.dll
bindepend.py               514 DEBUG    Skipping KERNEL32.dll dependency of mkl_rt.dll
bindepend.py               211 DEBUG    Analyzing c:\users\bjones\downloads\winpython-32bit-3.4.4.1qt5\python-3.4.4\Lib\site-packages\numpy\core\libmmd.dll
bindepend.py               514 DEBUG    Skipping KERNEL32.dll dependency of libmmd.dll
```

I'm not sure yet why it doesn't find all the dependencies. Still looking. `depend.bindepend.getfullnameof` at least generates the correct path on my PC.

**Comment by bjones1** [comment]
It looks like `mkl_rt.dll`, `libmmd.dll`, and `libifcoremd.dll` are found, but nothing else. Can someone recommend a good dependency walker tool? Neither [dependency walker](http://www.dependencywalker.com/) nor PE Explorer seemed to be able to find the additional dependencies. Or perhaps this needs to be added to a hook?

**Comment by codewarrior0** [comment]
> `Intel MKL FATAL ERROR: Cannot load mkl_intel_thread.dll`

I'm certain this means that the DLL in question is being loaded at runtime via `LoadLibrary` because the error message is unique to MKL. If it were loaded at load time, you would see a Python ImportError with "The specified DLL could not be found".

We can't scan for DLL loads done via `LoadLibrary`, so the only way out is a hook.

**Comment by bjones1** [comment]
Good point -- thanks. I'll look at some hooking.

**Comment by bjones1** [comment]
Closed by 08020fd185d23a9a316583adef82c5b94d10bd0e.
